applications/sink/zeromq-sink/src/test/java/org/springframework/cloud/stream/app/sink/zeromq/ZeroMqSinkInvalidConfigTests.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.sink.zeromq;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.bind.validation.BindValidationException;
+import org.springframework.boot.context.properties.bind.validation.ValidationErrors;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.cloud.fn.consumer.zeromq.ZeroMqConsumerProperties;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.validation.FieldError;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Tests for ZeroMqSink with invalid config.
+ *
+ * @author Daniel Frey
+ * @since 3.1.0
+ */
+public class ZeroMqSinkInvalidConfigTests {
+
+	@Test
+	public void testEmptyConnectionUrl() {
+		try {
+			AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+			TestPropertyValues.of("zeromq.consumer.connectUrl:").applyTo(context);
+			context.register(Config.class);
+			context.refresh();
+			fail("BeanCreationException expected");
```

Comment:
See `assertThatExceptionOfType()` API instead of this combination of `try..catch` and `fail`